File: Byte&Buy.AngularSPA/src/app/core/dto/checkout/checkout-response.ts
import { MoneyDto } from "../common/money-dto";
import { SelectListItemResponseInt } from "../common/select-list-item-response-int";
import { SellerGroup } from "./seller-group";

export interface CheckoutResponse {
  buyerFirstName: string;
  buyerLastName: string;
  buyerPhone: string;
  sellersGroups: SellerGroup[];
  avaliablePaymentMethods: SelectListItemResponseInt[];
  itemsCost: MoneyDto;
  tax: MoneyDto;
  totalCost: MoneyDto;
  canPlaceOrder: boolean;
}

File: Byte&Buy.Backend/src/ByteBuy.Core/DTO/Checkout/CheckoutResponse.cs
using ByteBuy.Core.DTO.Money;

namespace ByteBuy.Core.DTO.Checkout;

public record CheckoutResponse(
    string BuyerFirstName,
    string BuyerLastName,
    string BuyerPhone,
    IReadOnlyCollection<SellerGroup> SellersGroups,
    MoneyDto ItemsCost,
    MoneyDto Tax,
    MoneyDto TotalCost);
